### PR TITLE
Theme Designer: added color palette

### PR DESCRIPTION
**Pull request by emmayjiang**
Added a color palette preview to the theme designer.
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/87f555dfcbe007e701f80b56e3c69c50141e5466'>87f555dfcbe007e701f80b56e3c69c50141e5466</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/theme-designer/src/components/Palette/Palette.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+export const Palette: React.FC<PaletteProps> = props => {
+  const styles = useStyles();
+
+  return (
+    <div>
+      <Caption1>Generated Palette</Caption1>
+      <div className={mergeClasses(styles.root, props.className)}>
+        {getBrands(props.brandColors).map(brandKey => {
+          const brandColor = props.brandColors[brandKey];
+          if (brandKey > 80) {
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/Palette/Palette.tsx:41]: if the only difference is className below, you don't need to gate this entire return

className={brandKey > 80 ? styles.whiteText : ''}

Comment (emmayjiang) [reply packages/react-components/theme-designer/src/components/Palette/Palette.tsx:41]: omg that's so clean and nice

File: packages/react-components/theme-designer/src/components/Content/Content.tsx [hunk 2]
```diff
@@ -13,13 +17,34 @@ const useStyles = makeStyles({
     justifyContent: 'center',
     flexDirection: 'column',
     ...shorthands.padding('40px', '10%', '0px', '10%'),
+    gridRowGap: tokens.spacingVerticalXXXL,
   },
 });
 
+export const brandWeb: BrandVariants = {
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/Content/Content.tsx:24]: lets make a comment that this data is temporary and will eventually be pulled from current theme

File: packages/react-components/theme-designer/src/components/Demo/Demo.tsx [hunk 3]
```diff
@@ -204,13 +202,13 @@ export const Column3 = () => {
 export const Demo: React.FC<ContentProps> = props => {
   const styles = useStyles();
   return (
-    <FluentProvider theme={teamsLightTheme}>
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/Demo/Demo.tsx:207]: I take it back. The app will get a static theme (or maybe let users pick light or dark mode) but the Demo is going to get wrapped in the custom theme created by the user. So lets leave this provider in here, and make the theme a prop.

File: packages/react-components/theme-designer/src/components/Palette/Palette.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <div>
+      <Caption1>Generated Palette</Caption1>
+      <div className={mergeClasses(styles.root, props.className)}>
+        {getBrands(props.brandColors).map(brandKey => {
+          const brandColor = props.brandColors[brandKey];
+          return (
+            <div
+              key={brandKey}
+              className={brandKey <= 80 ? mergeClasses(styles.block, styles.whiteText) : styles.block}
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/Palette/Palette.tsx:44]: let's be explicit about each color. i.e. have a blackText classname and switch between those. I don't want the default color to change accidentally.

then you can write

```
mergeClasses(
  styles.block,
  brandKey <= 80 ? styles.whiteText : styles.blackText
);
```

**Review by micahgodbolt — APPROVED** [review]
A few suggestions but otherwise a great start!

File: packages/react-components/theme-designer/src/components/Palette/Palette.tsx (hunk 4 above)
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/Palette/Palette.tsx:44]: One future todo is that brandKey <= 80 is not a consistent way to check contrast. When we have dynamic themes, we'll need to use some 'getContrastRatio' function to determine when to switch over.

**Pull request merged**